stake-pool/program/src/processor.rs

```diff
@@ -941,70 +1047,241 @@ mod tests {
         // Create stake account
         let mut pool_info = create_stake_pool(fee);
 
-        let stake_account_key = Pubkey::new_unique();
-        let mut stake_account_account = Account::new(stake_balance, 100, &stake_program_id());
-        // TODO: Set stake account Withdrawer authority to pool_info.deposit_authority_key
+        let deposit_info = do_deposit(&mut pool_info, stake_balance);
 
-        // Create account to receive minted tokens
-        let (token_receiver_key, mut token_receiver_account) = create_token_account(
+        // Test stake pool balance
+        let state = State::deserialize(&pool_info.pool_account.data).unwrap();
+        match state {
+            State::Unallocated => panic!("Stake pool state is not initialized after deposit"),
```

Comment:
nit: Prefer to use the `matches!` macro instead https://doc.rust-lang.org/std/macro.matches.html